frontend/src/js/util/stringUtils.ts

```diff
@@ -1,3 +1,11 @@
 export function getLastPart(input: string, separator: string) {
-    return input.split(separator).slice(-1)[0];
+  return input.split(separator).slice(-1)[0];
+}
+
+export function removeLastUnmatchedQuote(input: string) {
```

Comment:
This is a good idea more generally. There are a few queries that can knock over our backend which should be pretty easy to detect. Anther example is `/` is considered an "begin regex" operator which is a bit mind blowning if you're a user I think.